### PR TITLE
Rincelent propose option to freely choose to flip normal

### PR DESCRIPTION
**Pull request by Rincelent**
Add a bunch of option to be able to revert front/back normal.

### CODE REVIEW

File: lib/extruder.js [hunk 1]
```diff
@@ -35,11 +35,15 @@ const extruder = {
         const res = {};
 
         if (options.frontMesh) {
-            res.frontMesh = extruder.getHorizontalGeom(horizontalPathsByDepth,innerPathsByDepth, [0], 0,true);
+            let invertFrontNormal = true;
+            if (options.doNotInvertFrontNormal) invertFrontNormal = false;
```
Comment (Krenodeno) [inline lib/extruder.js]: why not directly assign like this :
```js
const inverseFrontNormal = !options.doNotInvertFrontNormal || true;
```

Comment (Rincelent) [reply lib/extruder.js]: Even more simplified by !options.doNotInvertFrontNormal

File: lib/extruder.js [hunk 2]
```diff
@@ -35,11 +35,13 @@ const extruder = {
...
         if (options.frontMesh) {
-            res.frontMesh = extruder.getHorizontalGeom(horizontalPathsByDepth,innerPathsByDepth, [0], 0,true);
+            res.frontMesh = extruder.getHorizontalGeom(horizontalPathsByDepth,innerPathsByDepth, [0], 0, !options.doNotInvertFrontNormal);
             uvHelper.mapHorizontal(innerPathsByDepth, outerShape, res.frontMesh, options);
         }
         if (options.backMesh) {
-            res.backMesh = extruder.getHorizontalGeom(horizontalPathsByDepth, innerPathsByDepth, [horizontalPathsByDepth.length - 1], 0, false);
+            let invertBackNormal = false;
+            if (options.invertBackNormal) invertBackNormal = true;
+            res.backMesh = extruder.getHorizontalGeom(horizontalPathsByDepth, innerPathsByDepth, [horizontalPathsByDepth.length - 1], 0, invertBackNormal);
```
Comment (Krenodeno) [inline lib/extruder.js]: Same here :smile:

Comment (Rincelent) [reply lib/extruder.js]: Correct by ternaire.

**Comment by Krenodeno** [comment]
LGTM

File: lib/extruder.js [hunk 3]
```diff
@@ -197,6 +197,9 @@ const extruder = {
             mergeVerticalGeometries: true,
             doNotBuild: [],
             debug:false,
+            doNotInvertFrontNormal: false,
+            invertBackNormal: false,
+
```
Comment (aardvarksaurus) [inline lib/extruder.js]: This new empty line may be removed?

**Review by aardvarksaurus — CHANGES_REQUESTED** [review]

File: lib/extruder.js (hunk 3 above)
Comment (Krenodeno) [reply lib/extruder.js]: @aardvarksaurus fixed, good for you ?

Comment (aardvarksaurus) [reply lib/extruder.js]: Yep! Thx

Comment (aardvarksaurus) [reply lib/extruder.js]: Yep! Thx

**Comment by aardvarksaurus** [comment]
LGTM

**Review by aardvarksaurus — APPROVED** [review]

**Pull request merged**